skbio/stats/composition.py

```diff
@@ -2058,3 +2061,423 @@ def dirmult_ttest(
+                    "FeatureID": response_var,
+                    "Covariate": var_name,
+                    "Log2(FC)": float(results.summary().tables[1]["Coef."][var_name]),
+                    "CI(2.5)": float(results.summary().tables[1]["[0.025"][var_name]),
+                    "CI(97.5)": float(results.summary().tables[1]["0.975]"][var_name]),
+                    "pvalue": results.pvalues[var_name],
+                }
+            except Exception:
+                try:
+                    logfc = float(results.summary().tables[1]["Coef."][var_name])
+                except Exception:
+                    logfc = np.NaN
+
+                try:
+                    ci25 = float(results.summary().tables[1]["[0.025"][var_name])
+                except Exception:
+                    ci25 = np.NaN
+
+                try:
+                    ci975 = float(results.summary().tables[1]["0.975]"][var_name])
+                except Exception:
+                    ci975 = np.NaN
+
+                individial_results = {
+                    "FeatureID": response_var,
+                    "Covariate": var_name,
+                    "Log2(FC)": logfc,
+                    "CI(2.5)": ci25,
+                    "CI(97.5)": ci975,
+                    "pvalue": results.pvalues[var_name],
+                }
+
+            output.append(individial_results)
+
+    return (output, submodels, list_of_vars)
+
+
+def dirmult_lme(
+    formula,
+    data,
+    metadata,
+    groups=None,
+    reml=True,
+    method=None,
+    draws=128,
+    seed=None,
+    pseudocount=0.5,
+    p_adjust="holm",
+    fit_kwargs={},
+    **kwargs,
+):
+    r"""Fit a Dirichlet Multinomial linear mixed effects model.
+
+    The Dirichlet-multinomial distribution is a compound distribution that
+    combines a Dirichlet distribution over the probabilities of a multinomial
+    distribution. This distribution is used to model the distribution of
+    species abundances in a community.
+
+    To fit the linear mixed effect model we first fit a Dirichlet-multinomial
+    distribution for each sample, and then we compute the fold change and
+    *p*-value for each feature. The fold change is computed as the slopes
+    from the resulting model. Statistical tests are then performed on the posterior
+    samples, drawn from each Dirichlet-multinomial distribution. The
+    log-fold changes as well as their credible intervals, the *p*-values and
+    the multiple comparison corrected *p*-values are reported.
+
+    This function uses ``MixedLM`` module from
+    ``statsmodels.regression.mixed_linear_model``
+
+    Parameters
+    ----------
+    formula : str or generic Formula object
+        The formula specifying the model
+    data : DataFrame
+        The data for the model. data must define __getitem__ with the keys in
+        the formula terms args and kwargs are passed on to the model
+        instantiation. E.g., a numpy structured or rec array, a dictionary,
```

Comment:
The described allowed types appears different than the stated type expectation on line 2214